agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -3726,17 +3726,21 @@ bool slave_thread::autoconfig_wsc_parse_m2_encrypted_settings(
     // cConfigData constructor, but then parsing will fail since the length
     // will be calculated wrong. TLVF does not support parsing network byte order
     // without full swap, so keep this workaround for now (another future TLVF V2 feature)
-    WSC::cConfigData config_data(decrypted, datalen, true);
+    auto config_data = WSC::configData::parse(decrypted, datalen);
```

Comment:
I wonder if it is really worthwhile to add a configData class for this. We could just as well directly use AttrList, IMO. But OK, the code is there now, so why not.

Reply:
It is convenient to have a class for this, it was really easy to write since it is very similar to M1 and M2. Also, AttrList is abstract, so we would have needed to change that.
My original design was that AttrList will never be instantiated by itself, and for every list, there will be a derived class with simple getters, validation function, and configuration struct which are used to create and parse such lists, so this is how I went on with configData class.